barretenberg/cpp/pil/avm/gadgets/poseidon2_full.pil

```diff
@@ -0,0 +1,87 @@
+include "./poseidon2.pil";
+
+// Performs the poseidon2 full hash
+// It is **mostly** well-constrained
+namespace poseidon2_full(256);
+    pol commit clk;
+    // These are the inputs to be hashed this round, we hash chunks of 3
+    pol commit input_0;
+    pol commit input_1;
+    pol commit input_2;
+
+    // Output of the hash it is matched with the result of the last permutation round;
+    pol commit output;
+
+    pol commit sel_poseidon;
+    sel_poseidon * (1 - sel_poseidon) = 0;
+    sel_poseidon = execute_poseidon_perm + end_poseidon;
+    pol TWOPOW64 = 18446744073709551616;
+
+    pol commit input_len;
+    // Only used at the start of a new poseidon2 hash
+    pol IV = TWOPOW64 * input_len;
+
+    // Start of a poseidon2 computation
+    pol commit start_poseidon;
+    start_poseidon * (1 - start_poseidon) = 0;
+    // When we end a poseidon, the next row must naturally have a start_poseidon
+    sel_poseidon' * (1 - main.sel_first) * (start_poseidon' - end_poseidon) = 0;
```

Comment:
Exclusion on first execution row feels strange. Do not we start the first poseidon hash here?
Also, after the last poseidon hash, we also toggle the start_poseidon selector, which is also unexpected.

Maybe you can removed this constraint and writing a comment about the fact that input_len, start_poseidon need to be properly constrained.